scan/cli.go

```diff
@@ -216,6 +241,79 @@ func ScanCmd(c *cli.Context) error {
 	return commands.Exec(scanCmd)
 }
 
+// New Build-Scan command that works directly with Xray
+func BuildScan(c *cli.Context) error {
+	if c.NArg() > 2 {
+		return cliutils.PrintHelpAndReturnError("Wrong number of arguments.", c)
+	}
+	buildConfiguration := createBuildConfiguration(c)
+	if err := validateBuildConfiguration(c, buildConfiguration); err != nil {
+		return err
+	}
+
+	serverDetails, err := createServerDetailsWithConfigOffer(c)
+	if err != nil {
+		return err
+	}
+	err = validateXrayContext(c)
+	if err != nil {
+		return err
+	}
+	format, err := commandsutils.GetXrayOutputFormat(c.String("format"))
+	if err != nil {
+		return err
+	}
+	buildScanCmd := scancommands.NewBuildScanCommand().SetServerDetails(serverDetails).SetFailBuild(c.BoolT("fail")).SetBuildConfiguration(buildConfiguration).
+		SetIncludeVulnerabilities(c.Bool("vulnerabilities")).SetOutputFormat(format)
+	return commands.Exec(buildScanCmd)
+}
+
+func DockerScan(c *cli.Context) error {
+	return containerScan(c, containerutils.DockerClient)
+}
+
+func containerScan(c *cli.Context, containerManagerType containerutils.ContainerManagerType) error {
+	if c.NArg() != 1 {
+		return cliutils.PrintHelpAndReturnError("Wrong number of arguments.", c)
+	}
+	serverDetails, err := createServerDetailsWithConfigOffer(c)
+	if err != nil {
+		return err
+	}
+	format, err := commandsutils.GetXrayOutputFormat(c.String("format"))
+	if err != nil {
+		return err
+	}
+	containerScanCommand := scancommands.NewEmptyContainerScanCommand()
+	containerScanCommand.SetServerDetails(serverDetails).SetOutputFormat(format).SetProject(c.String("project")).
+		SetIncludeVulnerabilities(shouldIncludeVulnerabilities(c)).SetIncludeLicenses(c.Bool("licenses"))
+	if c.String("watches") != "" {
+		containerScanCommand.SetWatches(strings.Split(c.String("watches"), ","))
+	}
+	containerScanCommand.SetImageTag(c.Args().Get(0)).SetContainerManagerType(containerManagerType)
+	return commands.Exec(containerScanCommand)
+}
+
+// Returns build configuration struct using the params provided from the console.
+func createBuildConfiguration(c *cli.Context) *rtutils.BuildConfiguration {
```

Comment:
This function is now duplicated twice in the code. We should probably move it into ```cliutils/utils.go``` and have the two places use it.
The same goes for the below ```validateBuildConfiguration``` function.